catalyst/utils/hydra_config.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from omegaconf import DictConfig, OmegaConf
+
+
+def prepare_config(cfg: DictConfig) -> DictConfig:
+    """
+    Prepare config. Add required parameters.
+
+    Args:
+        cfg: (DictConfig) config
+
+    Returns:
+        DictConfig: config
+
+    """
+    OmegaConf.set_readonly(cfg, False)
+    OmegaConf.set_struct(cfg, False)
+
+    cfg.setdefault("vals", DictConfig({}))
+
+    cfg.setdefault("args", DictConfig({}))
+    cfg.args.setdefault("expdir", ".")
+    cfg.args.setdefault("resume", None)
+    cfg.args.setdefault("autoresume", None)
+    cfg.args.setdefault("seed", 42)
+    cfg.args.setdefault("distributed", os.getenv("USE_DDP", "0") == "1")
+    cfg.args.setdefault("apex", os.getenv("USE_APEX", "0") == "1")
+    cfg.args.setdefault("amp", os.getenv("USE_AMP", "0") == "1")
```

Comment:
Thanks for working on this!
While I understand you have backward compatibility requirements that might be in conflict with the suggestions here, I wanted to provide them anyway (at least as food for thought) :
I'd recommend utilizing config composition to get this instead of manually editing the config.
for example, the defaults list of the primary config can look like:
```yaml
defaults:
  - catalyst/config  # mandatory catalyst config (provided by catalyst)
  - foo: bar # user config groups
```
catalyst/config can contain everything needed by catalyst (what you are adding above).

A few extra points:
1. I am guessing this is designed to be backward compatible with the standard command line.
If possible, break away from that and utilize hierarchy to give catalyst a space of it's own where it wouldn't conflict with other frameworks/user config.
for example:
*catalst/config.yaml*:
```yaml
catalyst:
  resume: null
  seed: 42
  ...
```

2. Consider using Structured Configs to get config type safety, the above yaml file can be replaced with something like:
```python
@dataclass
class CatalystConf:
  resume: Optional[bool] = None
  seed: int = 42

ConfigStore.instance().store(group="catalst", name="config", node=CatalystConf)
```
(See the Structured Configs tutorial for more info).

Reply:
Thank you. Thinking about it.